README.md

```diff
@@ -2,25 +2,284 @@
+classes :: [ClassName] -> Prop
+
+-- | Attach a CSS class to the element
+class_ :: ClassName -> Prop
+```
+
+For example, to produce the following HTML:
+
+```html
+<div class="alert alert-warning alert-dismissible fade show" role="alert">
+  <strong>Holy guacamole!</strong> You should check in on some of those fields below.
+  <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+    <span aria-hidden="true">&times;</span>
+  </button>
+</div>
+```
+
+One would write the following Specular code:
+
+```purescript
+el "div" [classes ["alert", "alert-warning", "alert-dismissible", "fade", "show"], attr "role" "alert"] do
+  el_ "strong" $ text "Holy guacamole!"
+  text " You should check in on some of those fields below."
+  el "button" [class_ "close", attrs ("type":="button" <> "data-dismiss":="alert" <> "aria-label":="Close")] do
+    el "span" [attr "aria-hidden"  "true"] do
+      text "×"
+```
+
+#### Dynamic text, attributes and classes
+
+Most of the `Prop` constructors have their dynamic counterparts. As a convention, their names end in `D`. For example:
+
+```purescript
+-- | Attach dynamic attributes to the element.
+attrsD :: Dynamic Attrs -> Prop
+
+-- | Attach dynamic CSS classes to the element
+classesD :: Dynamic [ClassName] -> Prop
+```
+
+`text` also has a dynamic counterpart:
+
+```purescript
+-- | Create a text node whose text will reflect the value of the given Dynamic.
+dynText :: Dynamic String -> Widget Unit
+```
+
+For convenience, utilities for common cases are provided such as:
+
+```purescript
+attrWhenD :: Dynamic Boolean -> AttrName -> AttrValue -> Prop
+classWhenD :: Dynamic Boolean -> ClassName -> Prop
+```
+
+For example: assume you have `name :: Dynamic String`.
+The code:
+
+```purescript
+let isLong nm = String.length nm >= 5
+el "div" [class_ "name", classWhenD (isLong <$> name) "long"] do
+  text "Your name is: "
+  dynText name
+```
+
+when `name` has value `"Jan"`, would produce
+
+```html
+<div class="name">Your name is Jan</div>
+```
+
+whereas when `name` has value `"Titelitury"`, would produce
+
+```html
+<div class="name long">Your name is Titelitury</div>
+```
+
+#### Dynamic DOM structure
+
+Sometimes changing text and attributes is not enough. For that there's `withDynamic_`:
+
+```purescript
+withDynamic_ :: forall a. Dynamic a -> (a -> Widget Unit) -> Widget Unit
+```
+
+Whenever the Dynamic changes, it will re-render a new `Widget` based on the latest value.
+
+Example:
+
+```purescript
+-- Assume loading :: Dynamic Boolean
+
+withDynamic_ loading $
+  if _ then
+    el "div" [class_ "loading"] $ text "Loading..."
+  else
+    el_ "div" do
+      el_ "h1" $ text "Content"
+      el_ "p" $ text "Bla bla bla"
+```
+
+Warning: Re-rendering a whole DOM block on each change has performance implications. Use with care.
+
+#### Handling events
+
+```purescript
+-- | Connect a DOM event on the node to a Callback.
+on :: EventType -> Callback DOM.Event -> Prop
```

Comment:
Mention `on_`? Often we really don't care  about `DOM.Event` (eg. on `form.onSubmit`)

Reply:
Actually in `onsubmit` we do care about event; without `preventDefault` the page would reload.

Reply:
Right!

Reply:
btw, `onClickPreventDefault` (with unit callback) would be a useful shorthand.